sdk/python/feast/feature_store.py

```diff
@@ -1017,6 +1013,30 @@ def _augment_response_with_on_demand_transforms(
                     ] = GetOnlineFeaturesResponse.FieldStatus.PRESENT
         return OnlineResponse(GetOnlineFeaturesResponse(field_values=result_rows))
 
+    def _get_feature_views_to_use(
+        self,
+        features: Optional[Union[List[str], FeatureService]],
+        allow_cache=False,
+        hide_dummy_entity: bool = True,
+    ) -> List[FeatureView]:
+
+        passed_in_feature_views = (
+            {view.name: view for view in features.feature_views}
+            if isinstance(features, FeatureService)
+            else {}
+        )
+
+        all_feature_views = [
```

Comment:
This name `all_feature_views` seems inaccurate if we are filtering.

Reply:
the logic before may have been a bit confusing. The logic in the new commits should make it clear we're not actually filtering. All the FVs from the registry are present